memote-report-app/src/app/report-data.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { ResultCard } from './resultcard.model';
+import { TestResult } from './test-result.model';
+
+@Injectable()
+export class ReportDataService {
+  metaData: any;
+  allTests: TestResult[] = [];
+  scoredCard: Object;
+  statisticsCards: ResultCard[] = [];
+
+  constructor(private http:HttpClient){
+  }
+
+  public loadResults(): void{
+    // this.http.get("data/testData.json").subscribe(data => {this.rawReportData = data});
+    this.http.get("data/testData.json").subscribe(data => {this.convertResults(data)});
+  }
+
+  public byID(string){
+    return this.allTests.find(x => x.id === string);
+  }
+
+  public getString(object){
+    return JSON.stringify(object);
+  }
+
+  private convertResults(data:Object): void{
+    // Store each test result as a TestResult object in a central list.
+    for (let test of Object.keys(data["tests"])){
+      this.allTests.push(
+        new TestResult(
+          test,
+          data["tests"][test]['data'],
+          data["tests"][test]['duration'],
+          data["tests"][test]['message'],
+          data["tests"][test]['metric'],
+          data["tests"][test]['result'],
+          data["tests"][test]['summary'],
+          data["tests"][test]['title'],
+         data["tests"][test]['type']));
+    };
```

Comment:
@ChristianLieven Related comment: https://github.com/opencobra/memote/pull/271/files#r154901166

```typescript
new TestResult(test, data["tests"][test])
```

This looks more neat and maintainable to me, you don't need to worry about the order of arguments anymore.